kubernetes/resource_kubernetes_labels.go

```diff
@@ -0,0 +1,196 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sschema "k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/restmapper"
+)
+
+func resourceKubernetesLabels() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceKubernetesLabelsCreate,
+		ReadContext:   resourceKubernetesLabelsRead,
+		UpdateContext: resourceKubernetesLabelsUpdate,
+		DeleteContext: resourceKubernetesLabelsDelete,
+		Schema: map[string]*schema.Schema{
+			"api_version": {
+				Type:        schema.TypeString,
+				Description: "The apiVersion of the resource to label.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"kind": {
+				Type:        schema.TypeString,
+				Description: "The kind of the resource to label.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"metadata": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "The name of the resource.",
+							Required:    true,
+							ForceNew:    true,
+						},
+						"namespace": {
+							Type:        schema.TypeString,
+							Description: "The namespace of the resource.",
+							Optional:    true,
+							ForceNew:    true,
+						},
+					},
+				},
+			},
+			"labels": {
+				Type:        schema.TypeMap,
+				Description: "A map of labels to apply to the resource.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceKubernetesLabelsCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	d.SetId(buildIdWithVersionKind(metadata,
+		d.Get("api_version").(string),
+		d.Get("kind").(string)))
+	return resourceKubernetesLabelsUpdate(ctx, d, m)
+}
+
+func resourceKubernetesLabelsRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	conn, err := m.(KubeClientsets).DynamicClient()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	apiVersion := d.Get("api_version").(string)
+	kind := d.Get("kind").(string)
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	name := metadata.GetName()
+	namespace := metadata.GetNamespace()
+
+	// figure out which resource client to use
+	dc, err := m.(KubeClientsets).DiscoveryClient()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	agr, err := restmapper.GetAPIGroupResources(dc)
```

Comment:
Do we want to handle an error that potentially can be returned here? If so, then the same question for the line [141](https://github.com/hashicorp/terraform-provider-kubernetes/pull/1616/files#diff-73ee614c2215562c5d1ba3f2a74f548683458bab4735ff64730748458dd59c85R141).